File: reverseString.go
package main

import "fmt"

func main() {
	var str string = "piggip"
	var str2 string = ""
	for i := len(str) - 1; i >= 0; i-- {
		str2 += string(str[i])
	}
	fmt.Println(str2 == str)
}

File: reverseString.java
public class reverseString {
    public static void main(String[] args) {
        String str = "piggip";
        String rev = "";
        for(int i = str.length()-1; i>=0; i--){
            char c = str.charAt(i);
            rev+=c;
        }
        System.out.println(rev.equals(str));
    }
}
